src/scene.nim

```diff
@@ -0,0 +1,93 @@
+import geometry, hdrimage, shapes
+
+from std/sequtils import concat, foldl, toSeq, filterIt, mapIt
+from std/algorithm import sorted
+
+
+type ShapeHandler* = tuple[shape: Shape, transformation: Transformation]
+    
+proc getAABB*(handler: ShapeHandler): Interval[Point3D] {.inline.} =
+    getAABB(handler.shape.getVertices.mapIt(apply(handler.transformation, it)))
+
+proc getLocalAABB*(refSystem: ReferenceSystem, handler: ShapeHandler): Interval[Point3D] {.inline.} =
+    let localTransformation = newComposition(handler.transformation, newTranslation(refSystem.origin).inverse)
+    getAABB(handler.shape.getVertices.mapIt(refSystem.coeff(apply(localTransformation, it)).Point3D))
+
+
+type
+    SceneNodeKind* = enum 
+        nkBranch, nkLeaf
+
+    # BranchNodeKind* = enum 
```

Comment:
Do you want to create different tree types? That would be cool, but I feel like we need to have this one working before doing so.